app/src/main/java/nl/hogeschoolrotterdam/projectb/LocationEditActivity.java

```diff
@@ -0,0 +1,152 @@
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+import com.google.android.gms.maps.*;
+import com.google.android.gms.maps.model.CameraPosition;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import nl.hogeschoolrotterdam.projectb.MainActivity;
+import nl.hogeschoolrotterdam.projectb.MemoryDetailActivity;
+import nl.hogeschoolrotterdam.projectb.R;
+import nl.hogeschoolrotterdam.projectb.data.Database;
+import nl.hogeschoolrotterdam.projectb.data.Memory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by maartendegoede on 20/03/2019.
+ * Copyright © 2019 Anass El Mahdaoui, Hicham El Marzgioui, Michaël van Asperen, Wesley de Man, Maarten de Goede all rights reserved.
+ */
+public class LocationEditActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+    GoogleMap mGoogleMap;
+    MapView mMapView;
+    private EditText mSearchText;
+    private static final String TAG = "Location";
+    private Button button;
+    LatLng latLng;
+
+
+    private void inity() {
+        mGoogleMap.setOnMapLongClickListener(new GoogleMap.OnMapLongClickListener(){
+
+            @Override
+            public void onMapLongClick(LatLng Longl) {
+                latLng = Longl;
+                setMarker(latLng);
+
+            }
+        });
+        mSearchText = (EditText) findViewById(R.id.input_search);
+        mSearchText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView textView, int actionId, KeyEvent event) {
+                if (actionId == EditorInfo.IME_ACTION_SEARCH
+                        || actionId == EditorInfo.IME_ACTION_DONE
+                        || event.getAction() == KeyEvent.ACTION_DOWN
+                        || event.getAction() == KeyEvent.KEYCODE_ENTER) {
+                    geoLocate();
+                }
+                return false;
+            }
+        });
+    }
+
+    private void geoLocate() {
+        String searchingString = mSearchText.getText().toString();
+        Geocoder geocoder = new Geocoder(this);
+        List<Address> list = new ArrayList<>();
+        try {
+            list = geocoder.getFromLocationName(searchingString, 1);
+
+
+        } catch (IOException e) {
+        }
+        if (list.size() > 0) {
+            Address address = list.get(0);
+            Log.d(TAG, "Locate adress:" + address.toString());
+            //Toast.makeText(this, address.toString(), Toast.LENGTH_SHORT).show();
+            CameraPosition search = CameraPosition.builder().target(new LatLng(address.getLatitude(),
```

Comment:
Shouldn't this `CamaeraPosition` thing be moved to the `setMarker` method?

Reply:
The camera position should always be moved to the current marker, so if you move these 2 lines to `setMarker`, it'll always happen.

Reply:
done